cli/medperf/commands/dataset/create.py

```diff
@@ -150,22 +151,33 @@ def run_cube_tasks(self):
 
         # Run the tasks
         self.ui.text = "Running preparation step..."
-        self.cube.run(
-            self.ui, task="prepare", timeout=prepare_timeout, **prepare_params
-        )
-        self.ui.print("> Cube execution complete")
+        try:
+            self.cube.run(
+                self.ui, task="prepare", timeout=prepare_timeout, **prepare_params
+            )
+            self.ui.print("> Cube execution complete")
 
-        self.ui.text = "Running sanity check..."
-        self.cube.run(
-            self.ui, task="sanity_check", timeout=sanity_check_timeout, **sanity_params
-        )
-        self.ui.print("> Sanity checks complete")
+            self.ui.text = "Running sanity check..."
+            self.cube.run(
+                self.ui,
+                task="sanity_check",
+                timeout=sanity_check_timeout,
+                **sanity_params,
+            )
+            self.ui.print("> Sanity checks complete")
 
-        self.ui.text = "Generating statistics..."
-        self.cube.run(
-            self.ui, task="statistics", timeout=statistics_timeout, **statistics_params
-        )
-        self.ui.print("> Statistics complete")
+            self.ui.text = "Generating statistics..."
+            self.cube.run(
+                self.ui,
+                task="statistics",
+                timeout=statistics_timeout,
+                **statistics_params,
+            )
+            self.ui.print("> Statistics complete")
+        except RuntimeError as e:
+            cleanup_paths = [out_datapath, out_labelspath]
```

Comment:
also `out_statistics_path`.
Maybe just use `self.out_path`?

Reply:
Agreed, that would be easier!